metrics_test.go

```diff
@@ -426,3 +429,166 @@ func TestMetrics_ErrorCountShouldDecreaseIfSuccessful(t *testing.T) {
 	assert.Equal(float64(0), client.metrics.errors)
 	assert.Nil(err, "Client should close without a problem")
 }
+
+func clientDataMatcher() func(req *http.Request, ereq *gock.Request) (bool, error) {
+	defaultStrategies := []string{
+		"default",
+		"applicationHostname",
+		"gradualRolloutRandom",
+		"gradualRolloutSessionId",
+		"gradualRolloutUserId",
+		"remoteAddress",
+		"userWithId",
+		"flexibleRollout",
+	}
+	return func(req *http.Request, ereq *gock.Request) (bool, error) {
+		var data ClientData
+		err := json.NewDecoder(req.Body).Decode(&data)
+		if err != nil {
+			return false, err
+		}
+
+		if data.Started.IsZero() {
+			return false, nil
+		}
+
+		expectedData := ClientData{
+			AppName:          mockAppName,
+			InstanceID:       mockInstanceId,
+			SDKVersion:       fmt.Sprintf("%s:%s", clientName, clientVersion),
+			Strategies:       defaultStrategies,
+			Interval:         0,
+			PlatformVersion:  runtime.Version(),
+			PlatformName:     "go",
+			YggdrasilVersion: nil,
+			SpecVersion:      specVersion,
+		}
+
+		return data.AppName == expectedData.AppName &&
+			data.InstanceID == expectedData.InstanceID &&
+			data.SDKVersion == expectedData.SDKVersion &&
+			compareStringSlices(data.Strategies, expectedData.Strategies) &&
+			data.Interval == expectedData.Interval &&
+			data.PlatformVersion == expectedData.PlatformVersion &&
+			data.PlatformName == expectedData.PlatformName &&
+			data.YggdrasilVersion == expectedData.YggdrasilVersion &&
+			data.SpecVersion == expectedData.SpecVersion, nil
+	}
+}
+
+func compareStringSlices(a, b []string) bool {
+	if len(a) != len(b) {
+		return false
+	}
+	for i := range a {
+		if a[i] != b[i] {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
These are helper functions, not tests. Maybe we should move them to the top of the file or somewhere else that's not here?

Reply:
Why not here, they're used immediately after?

Reply:
Simply readability / preference. I don't have strong feelings either way, so I'm happy with keeping this.